test/integration/framework/etcd_utils.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// +build integration
+
+package framework
+
+import (
+	"fmt"
+	"math/rand"
+
+	"github.com/coreos/go-etcd/etcd"
+	"github.com/golang/glog"
+)
+
+// If you need to start an etcd instance by hand, you also need to insert a key
+// for this check to pass (*any* key will do, eg:
+//curl -L http://127.0.0.1:4001/v2/keys/message -XPUT -d value="Hello world").
+func init() {
+	RequireEtcd()
+}
+
+func NewEtcdClient() *etcd.Client {
+	return etcd.NewClient([]string{})
+}
+
+func RequireEtcd() {
+	if _, err := NewEtcdClient().Get("/", false, false); err != nil {
+		glog.Fatalf("unable to connect to etcd for testing: %v", err)
+	}
+}
+
+func WithEtcdKey(f func(string)) {
+	prefix := fmt.Sprintf("/test-%d", rand.Int63())
+	defer NewEtcdClient().Delete(prefix, true)
+	f(prefix)
+}
+
+func DeleteAllEtcdKeys() {
```

Comment:
// TODO: Instead of sprinkling calls to this throughout the code, adjust the prefix in etcdtest package; then just delete everything once at the end of the test run.